hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -165,12 +170,14 @@ public void initServices() {
                 partitionManager.registerDataListener(this);
                 partitionManager.registerTriggerListener(this);
             }
-            /*
-                A reload() operation means that all triggers from the backend should be reloaded into the AlertsEngine
-                memory. In a distributed environment, the node that execute the reload() operation loads the triggers
-                assigned to it and notify other nodes to load rest of the triggers.
-             */
-            reload();
+            executor.submit(() -> {
+                /*
+                    A reload() operation means that all triggers from the backend should be reloaded into
+                    the AlertsEngine memory. In a distributed environment, the node that execute the reload()
+                    operation loads the triggers assigned to it and notify other nodes to load rest of the triggers.
+                 */
+                reload();
```

Comment:
Not with that code, as it is executed on a Singleton marked a Startup.
But we have for historical reasons a /reload endpoint that it can be removed safely in the future.
I don't remember any test-case using the REST endpoint for reload it was there for the first primitive tests.